### PR TITLE
Ticket 848 - Uploading a shapefile logic

### LINKED ISSUE #848: Uploading a shapefile UI and logic
- [x] User is able to EDIT/SAVE/DELETE uploaded geometry (same functionality as with the draw polygon widget) #972 
- [x] EDIT/DELETE button UI matches comps #975 
- [x] Shapefile upload component basic styling #851
- [x] Shapefile upload component exists in pen widget #849
- [x] Shapefile upload component exists in analysis (default) tab #854 
- [x] User can upload geojson file #859 
- [x] When uploaded, make API call to GFW endpoint to convert https://production-api.globalforestwatch.org/v1/ogr/convert ( PROD in `upload.js`) it returns GEOJSON #859 
- [x] Then we convert a shapefile to geojson, then convert it to arcgis (we can utilize util functions from PROD `arcgis-to-jsonjs` #859 
- [x] We draw that shape on the map. And zoom to it. #859 
- [x] Opens analysis tab and populates data tab with content #860
- [x] Conditionally closes modal #860
- [x] Clear previous graphic before adding a graphic to the mapview #866 
- [x] Register geometry with the geostore and add it to the active features array #869

### PR DESCRIPTION
**Pull request by eyosef**
This PR registers the geoStoreID of an uploaded file (shapefile or geoJSON)

Fixes part of #848 

What it accomplishes;
- updates helper function logic of `registerGeometry()` to account for `wkid` property
- registers the geometry of the uplaoded shapefile to generate a geostore ID
- updates redux store property `activeFeatures`

What it does not accomplish;
- N/A

### CODE REVIEW

**Review by vaidotasp — CHANGES_REQUESTED** [review]
Can we account for multiple shapefiles/geojson uploads? What I mean that if user uploads one geojson, and then proceeds to upload a second one. The first one should be cleared (cleared from the map, cleared from redux) and only the second should be present.

File: src/js/components/sharedComponents/UploadFile.tsx [hunk 1]
```diff
@@ -89,10 +96,37 @@ const UploadFile = (): JSX.Element => {
...
+        })
+        .catch(e =>
+          console.log('error in registerGeometry() in onDropFile()', e)
+        );
+
+      mapController.processGeojson(arcGISResults);
 
-      const results = geojsonToArcGIS(featureCollection.data.attributes);
-      mapController.processGeojson(results);
       dispatch(toggleTabviewPanel(true));
```
Comment (vaidotasp) [inline src/js/components/sharedComponents/UploadFile.tsx:153]: I am not sure why we need that action? `selectActiveTab` is already opening the tab?

**Review by vaidotasp — CHANGES_REQUESTED** [review]

File: src/js/components/sharedComponents/UploadFile.tsx (hunk 1 above)
Comment (eyosef) [reply src/js/components/sharedComponents/UploadFile.tsx:153]: @vaidotasp this accounts for when the user uploads a shapefile/geojson via the pen widget

**Comment by eyosef** [comment]
@vaidotasp just following up with your feedback [here](https://github.com/wri/gfw-mapbuilder/pull/869#pullrequestreview-368057053). Looks like I just needed to merge the latest `clean-slate` branch into this feature branch, since clearing previous shapefile/geojson uploads was resolved previously. If that's not the case, just let me know, thanks!

File: src/js/components/sharedComponents/UploadFile.tsx (hunk 1 above)
Comment (vaidotasp) [reply src/js/components/sharedComponents/UploadFile.tsx:153]: 👍

File: src/js/components/sharedComponents/UploadFile.tsx [hunk 2]
```diff
@@ -89,13 +97,57 @@ const UploadFile = (): JSX.Element => {
         body: formData
       })
         .then(response => response.json())
-        .catch(e => console.log('error in onDropFile()', e));
+        .catch(e => console.log('fetching error in onDropFile()', e));
+
+      const arcGISResults = geojsonToArcGIS(featureCollection.data.attributes);
+
+      await Promise.all(
```
Comment (vaidotasp) [inline src/js/components/sharedComponents/UploadFile.tsx]: This `await` looks to be doing nothing, we are calling `.then` on Promise.all anyway?

File: src/js/components/sharedComponents/UploadFile.tsx [hunk 3]
```diff
@@ -89,13 +97,57 @@ const UploadFile = (): JSX.Element => {
...
+            const shapeFileFeatures: LayerFeatureResult = {
+              layerID: 'upload_file_features',
+              layerTitle: 'Upload File Features',
+              sublayerID: null,
+              sublayerTitle: null,
+              features: graphics
+            };
+
+            oldActiveFeatures.push(shapeFileFeatures);
+            dispatch(setActiveFeatures(oldActiveFeatures));
```
Comment (vaidotasp) [inline src/js/components/sharedComponents/UploadFile.tsx]: This logic is leaving artifacts of previous upload. In the data panel if you upload a second file after the first, they both show up and it has some regressions, we need to clean out the activeFeature array with only single item (our shapeFileFeatures). Also from feature result we do not need anything else except attributes and geometry. Check out suggestion below.

```suggestion
            const graphics = mapController.generateGraphics(arcGISResults);

            const shapeFileFeatures: LayerFeatureResult = {
              layerID: 'upload_file_features',
              layerTitle: 'Upload File Features',
              sublayerID: null,
              sublayerTitle: null,
              features: graphics.map((g: __esri.Graphic) => {
                return { attributes: g.attributes, geometry: g.geometry };
              })
            };

            dispatch(setActiveFeatures([shapeFileFeatures]));
```

File: src/js/store/mapview/types.ts [hunk 4]
```diff
@@ -86,6 +86,7 @@ export interface LayerFeatureResult {
   sublayerTitle: string | null;
   sublayerID: string | null;
   features: FeatureResult[];
+  geoStoreID?: string;
```
Comment (vaidotasp) [inline src/js/store/mapview/types.ts]: May not need this anymore?
```suggestion

```

**Review by vaidotasp — CHANGES_REQUESTED** [review]

File: src/js/helpers/MapGraphics.ts [hunk 5]
```diff
@@ -32,16 +34,44 @@ export function setNewGraphic({
...
+  };
 
   allFeatures.forEach((feature: FeatureResult) => {
-    const symbol = (feature.geometry as any).rings
-      ? getCustomSymbol()
-      : polygonOrPointSymbol(feature.geometry.type);
+    const isPolygon =
+      (feature.geometry as any).rings || feature.geometry.type === 'polygon'
+        ? true
+        : false;
```
Comment (vaidotasp) [inline src/js/helpers/MapGraphics.ts]: not sure that we need `? true : false` that logic already evaluates to appropriate boolean.

File: src/js/helpers/MapGraphics.ts [hunk 6]
```diff
@@ -32,16 +34,44 @@ export function setNewGraphic({
...
+    }
+  };
+
+  const setGeometry = (symbolType: string, geometry: __esri.Geometry): any => {
+    switch (symbolType) {
+      case 'polygon':
+        return new Polygon(geometry);
+      case 'point':
+        return new Point(geometry);
+    }
```
Comment (vaidotasp) [inline src/js/helpers/MapGraphics.ts]: if we have a switch statement, lets add default case too. it can try and return a Polygon?

**Review by vaidotasp — CHANGES_REQUESTED** [review]

**Review by vaidotasp — APPROVED** [review]

**Pull request merged**